### PR TITLE
Implementation of maximum elution time filter; Devcontainer.json bug fix

### LINKED ISSUE #62: Optimize the devcontainer
- [ ] Log on as local user instead of root to avoid write issues etc.
- [ ] Foward git credentials reliably
- [ ] install vscDebugger package

### LINKED ISSUE #231: Implement maximum elution time filter
The current implementation of the `recetox-aplcms` package includes a filter that removes peaks with **short elution profiles**, based on a minimum retention time (RT) threshold combining with a signal presence threshold. However, it does **not currently filter out peaks with excessively long elution profiles**, which may represent noise, baseline drift, or poorly defined features.

This issue proposes adding a **maximum elution time filter** as part of the noise removal step, using a configurable parameter to define the upper bound of acceptable RT span.

The relevant source code for this is located in the ```run_filter``` function defined in [R/run_filter.R](..tree/master/R/run_filter.R).

---

### Motivation

- **Long elution profiles** may indicate non-specific signals, unresolved baseline, or artifacts.
- Filtering these out can improve the **specificity and quality** of detected features.
- This complements the existing **minimum elution time filter**, creating a more versatile noise removal strategy.

---

### Proposed Implementation

1. **Calculate elution profile length** for each EIC (Extracted Ion Chromatogram) group:
   - Use the ```grps``` column or ```group_number```, depending on whether this is implemented as pre- min elution time filter or post.
   - Use the difference between the **minimum and maximum RT values** within each group as the elution time
2. **Introduce a new parameter** (e.g., `max_elution_time`) to define the upper threshold for acceptable RT span.
3. **Add filtering logic** to exclude peaks where:
   ```r
   (max(rt) - min(rt)) > max_elution_time
   ```
4. **Integrate this filter** into the existing noise removal step, ensuring it runs after or alongside the minimum elution time filter.
5. **Document the new filter** in the package reference and usage examples.

---

### Tasks

- [x] Add `max_elution_time` parameter to relevant functions.
- [x] Implement the filter at the same place as the `min_elution_time` filtering in https://github.com/RECETOX/recetox-aplcms/blob/85d8b9fc49aa57765e08312064bd126a3059fc39/R/run_filter.R#L108
- [x] Update and expand unit tests, also to cover edge cases (e.g., borderline RT spans).
- [ ] Add documentation and examples.

---

### Related Context

This enhancement builds on the existing RT-based filtering logic and supports better preprocessing for downstream steps such as alignment and quantification. It may also help reduce false positives in datasets with high baseline noise or low chromatographic resolution.

### PR DESCRIPTION
**Pull request by luciakostialova**
- maximum elution time filter parameter was added into run_filter function. Unit tests were added.
- group_threshold parameter from remove_noise function - syntax was corrected
- devcontainer.json was updated to fix bugs

### CODE REVIEW

File: R/run_filter.R [hunk 1]
```diff
@@ -104,8 +106,9 @@ run_filter <- function(newprof,
   # ordered by mz and grps data that are inside unigrps
   newprof <- dplyr::filter(newprof, grps %in% uniq_grp) |> dplyr::arrange(grps, mz)
 
+  # compute elution profile length for each EIC group, and filter accordingly
   results <- dplyr::group_by(newprof, grps) |>
-    dplyr::filter(n() >= min_count_run && abs(span(rt)) >= min_run) |>
+    dplyr::filter(n() >= min_count_run && abs(span(rt)) >= min_run && abs(span(rt)) <= max_run) |>
```
Comment (hechth) [inline R/run_filter.R]:
```suggestion
    dplyr::filter(n() >= min_count_run && span(rt) >= min_run && span(rt) <= max_run) |>
```

**Review by hechth — APPROVED** [review]
